frontend-client/src/components/courseOverview.tsx

```diff
@@ -4,37 +4,39 @@ import SearchIcon from '@mui/icons-material/Search';
 
 
 
-const CourseOverview = () => {
+class CourseOverview extends React.Component {
+    render() {
 
-    const [bar, switchBar] = useState(1);
+        const [bar, switchBar] = useState(1);
```

Comment:
we should use this.state instead of hooks , thats not how we use the state with class components

Reply:
Ready for review.